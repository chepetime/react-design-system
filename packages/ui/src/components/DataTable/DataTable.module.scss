@use "./../../styles/colors.scss";
@use "./../../styles/mixins";

:root {
}

:global :root .dark {
}

.DataTable {
  all: revert;
  box-sizing: border-box;

  background: red;

  @include mixins.respond-to("md") {
    background-color: cyan;
  }
  @include mixins.respond-to("lg") {
    background-color: green;
  }
  @include mixins.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
