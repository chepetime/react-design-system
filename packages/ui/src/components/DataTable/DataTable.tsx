import React from "react";
import classnames from "classnames";
import css from "./DataTable.module.scss";

export interface DataTableProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const DataTable = React.forwardRef<HTMLDivElement, DataTableProps>(
  (props: DataTableProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.DataTable)} ref={ref}>
        <p>DataTable component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

DataTable.displayName = "DataTable";
