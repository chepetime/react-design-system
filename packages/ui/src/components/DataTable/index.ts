export { DataTable } from "./DataTable";
export type { DataTableProps } from "./DataTable";