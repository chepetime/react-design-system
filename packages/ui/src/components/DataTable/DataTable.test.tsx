import React from "react";import { render } from "@testing-library/react";
import { DataTable } from "./DataTable";

describe("DataTable", () => {
  test("renders the DataTable component", () => {
    render(<DataTable />);
  });
});
