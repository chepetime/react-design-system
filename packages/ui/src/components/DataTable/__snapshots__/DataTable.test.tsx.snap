// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable matches snapshot 1`] = `
<div>
  <div
    class="DataTable"
  >
    <p>
      DataTable component working!
    </p>
    <div />
  </div>
</div>
`;
